bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/ZkLedgerUnderreplicationManager.java

```diff
@@ -737,6 +737,40 @@ public boolean isLedgerReplicationEnabled()
         }
     }
 
+    @Override
+    public void notifyLedgerReplicationStatusChanged(final GenericCallback<Void> cb)
+            throws ReplicationException.UnavailableException {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("notifyLedgerReplicationStatusChanged()");
+        }
+        Watcher w = new Watcher() {
+            @Override
+            public void process(WatchedEvent e) {
+                if (e.getType() == Watcher.Event.EventType.NodeDeleted) {
+                    LOG.info("aLedgerReplication is enabled externally through Zookeeper, "
+                            + "since DISABLE_NODE ZNode is deleted");
+                    cb.operationComplete(0, null);
+                } else if (e.getType() == Watcher.Event.EventType.NodeCreated) {
+                    LOG.info("bLedgerReplication is disabled externally through Zookeeper, "
+                            + "since DISABLE_NODE ZNode is created");
+                    cb.operationComplete(1, null);
+                }
+            }
+        };
+        try {
+            zkc.addWatch(basePath + '/'
```

Comment:
We must implement this for every supported Metadata driver 
What happens if the driver doesn't implement it?

Reply:
If this, We also have this problem  currently.  see `notifyLedgerReplicationEnabled `